src/lsst_efd_client/efd_helper.py

```diff
@@ -671,3 +868,382 @@ def _parse_schema(topic, schema):
+            fields,
+            start,
+            end,
+            is_window,
+            index,
+            convert_influx_index,
+            use_old_csc_indexing,
+        )
+        ret = self._influx_client.do_query(query, convert_influx_index)
+        if ret.empty and not self._is_topic_valid(topic_name):
+            raise ValueError(f"Topic {topic_name} not in EFD schema")
+        return ret
+
+    def select_top_n(
+        self,
+        topic_name,
+        fields,
+        num,
+        time_cut=None,
+        index=None,
+        convert_influx_index=False,
+        use_old_csc_indexing=False,
+    ):
+        query = EfdClientTools.build_select_top_n_query(
+            topic_name,
+            fields,
+            num,
+            self._influx_client.db_name,
+            time_cut=None,
+            index=None,
+            convert_influx_index=False,
+            use_old_csc_indexing=False,
+        )
+
+        ret = self._influx_client.do_query(query, convert_influx_index)
+
+        if ret.empty and not self._is_topic_valid(topic_name):
+            raise ValueError(f"Topic {topic_name} not in EFD schema")
+        return ret
+
+    async def select_packed_time_series(
+        self,
+        topic_name,
+        base_fields,
+        start,
+        end,
+        is_window=False,
+        index=None,
+        ref_timestamp_col="cRIO_timestamp",
+        ref_timestamp_fmt="unix_tai",
+        ref_timestamp_scale="tai",
+        convert_influx_index=False,
+        use_old_csc_indexing=False,
+    ):
+        fields = self.get_fields(topic_name)
+        field_list = EfdClientTools.make_fields(fields, base_fields)
+        result = self.select_time_series(
+            topic_name,
+            field_list
+            + [
+                ref_timestamp_col,
+            ],
+            start,
+            end,
+            is_window=is_window,
+            index=index,
+            convert_influx_index=convert_influx_index,
+            use_old_csc_indexing=use_old_csc_indexing,
+        )
+        return EfdClientTools.merge_packed_time_series(
+            result,
+            base_fields,
+            ref_timestamp_col,
+            ref_timestamp_fmt,
+            ref_timestamp_scale,
+        )
+
+    def _is_topic_valid(self, topic: str) -> bool:
+        # A helper function that check if the specified topic is in the schema.
+        # A topic is valid and returns `True` if it is in the cached list of
+        # topics. Any other case returns `False`.
+        existing_topics = self.get_topics()
+        return topic in existing_topics
+
+    def get_schema(self, topic):
+        with aiohttp.ClientSession() as http_session:
+            registry_api = RegistryApi(
+                session=http_session,
+                url=self._influx_client.schema_registry_url,
+            )
+            schema = registry_api.get_schema_by_subject(f"{topic}-value")
+        return EfdClientTools.parse_schema(topic, schema)
```

Comment:
`get_schema_by_subject` is an async method, so this code won't work. kafkit only supports async, so you'll need to replace this with code that doesn't use kafkit in the sync client or drop this functionality from the sync client.

Reply:
Thanks a lot for the review, I made all the required changes. Yes, it looks like some of the code didnt really added extra functionality and could be removed, now its much light and easy to follow. Only pending the get_schema for the sync client, for now its removed, it may be added later,

Regards

Reply:
Hello @rra, I also modified the get_schema on the synchronous client. Hope the PR can get approved soon! 
Thanks and regards